src/jit/lir.cpp

```diff
@@ -1110,6 +1113,28 @@ bool LIR::Range::CheckLIR(Compiler* compiler, bool checkUnusedValues) const
     // Make sure the range itself is valid.
     assert(IsValid());
 
+    // Make sure the list is well-formed: no node appears more than once (no circularities). Insert each node in a set
```

Comment:
Did you actually run in to circularities? It shouldn't be possible to construct such lists using the `Insert*`/`Remove` methods (though it is possible via manual mutations of `gtNext`/`gtPrev`)